openmp/runtime/src/kmp_utils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * kmp_utils.h -- Utilities that used internally
+ */
+
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef __KMP_UTILS_H__
+#define __KMP_UTILS_H__
+
+#include <cstddef>
+
+/// A simple pure header implementation of VLA that aims to replace uses of
+/// actual VLA, which can cause compile warning.
+/// Similar to the actual VLA, we don't check boundary (for now), so we will not
+/// store the size. We can always revise it later.
+template <typename T> class SimpleVLA final {
+  T *Data = nullptr;
+
+public:
+  SimpleVLA() = delete;
+  SimpleVLA(const SimpleVLA &) = delete;
+  SimpleVLA(SimpleVLA &&) = default;
+  SimpleVLA &operator=(const SimpleVLA &) = delete;
+  SimpleVLA &operator=(SimpleVLA &&) = default;
```

Comment:
Defaulted move functions are still problematic. Once we move constructs a new `SimpleVLA` from the old one, the old one will hold the data, and double-free will probably happen.

Can we simply reuse `unique_ptr<T[]>` and `make_unique`?

Reply:
I'm not sure if the smart pointer will introduce the dependence on any form of C++ library since we can't do that.

Reply:
I suppose it's OK to use `unique_ptr`, given the current stategy is already using `operator new[]` and `operator delete[]` form the C++ standard library.

Reply:
FWIW, we already use `operator new[]` in `libomp` but it doesn't look like it actually uses things from C++ standard library.

Reply:
I removed the move functions because I don't think that is allowed by actual VLA.